GUI/src/main/kotlin/models/AppState.kt

```diff
@@ -21,6 +21,9 @@ data class AppState(
     var maskPath: String = getPath("mask.png"),
     var gapOpenPenalty: Double = 0.2,
     var gapExtendPenalty: Double = -0.8,
+    var prevGapOpenPenalty: Double = 0.2,
+    var prevExtendPenalty: Double = -0.8,
+    var prevMaskPath: String = getPath("mask.png"),
```

Comment:
this is not how a state is used. If you want a backup of all values, just save a copy of the state, and override the current state with a backup if you want to restore values